src/main/scala/com/microsoft/hyperspace/index/rules/PlanUtils.scala

```diff
@@ -0,0 +1,190 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.microsoft.hyperspace.index.rules
+
+import scala.util.Try
+
+import org.apache.spark.sql.catalyst.expressions.{Alias, Attribute, AttributeReference, Expression, GetStructField}
+import org.apache.spark.sql.catalyst.plans.logical.{Filter, LogicalPlan, Project}
+import org.apache.spark.sql.types.{DataType, StructType}
+
+import com.microsoft.hyperspace.util.SchemaUtils
+
+object PlanUtils {
+
+  /**
+   * Returns true if the given project is a supported project. If all of the registered
+   * providers return None, this returns false.
+   *
+   * @param project Project to check if it's supported.
+   * @return True if the given project is a supported relation.
+   */
+  def isSupportedProject(project: Project): Boolean = {
+    val containsNestedFields =
+      SchemaUtils.hasNestedFields(project.projectList.flatMap(extractNamesFromExpression))
+    var containsNestedChildren = false
+    project.child.foreach {
+      case f: Filter =>
+        containsNestedChildren = containsNestedChildren || {
+          SchemaUtils.hasNestedFields(
+            SchemaUtils.unescapeFieldNames(extractNamesFromExpression(f.condition).toSeq))
+        }
+      case _ =>
+    }
+    containsNestedFields || containsNestedChildren
+  }
+
+  /**
+   * Returns true if the given filter is a supported filter. If all of the registered
+   * providers return None, this returns false.
+   *
+   * @param filter Filter to check if it's supported.
+   * @return True if the given project is a supported relation.
+   */
+  def isSupportedFilter(filter: Filter): Boolean = {
+    val containsNestedFields =
+      SchemaUtils.hasNestedFields(extractNamesFromExpression(filter.condition).toSeq)
+    containsNestedFields
+  }
+
+  /**
+   * Given an expression it extracts all the field names from it.
+   *
+   * @param exp Expression to extract field names from
+   * @return A set of distinct strings representing the field names
+   *         (ie: `Set(nested.field.id, nested.field.other)`)
+   */
+  def extractNamesFromExpression(exp: Expression): Set[String] = {
+    exp match {
+      case AttributeReference(name, _, _, _) =>
+        Set(s"$name")
+      case otherExp =>
+        otherExp.containsChild.flatMap {
+          case g: GetStructField =>
+            Set(s"${getChildNameFromStruct(g)}")
+          case e: Expression =>
+            extractNamesFromExpression(e).filter(_.nonEmpty)
+          case _ => Set.empty[String]
+        }
+    }
+  }
+
+  /**
+   * Given a nested field this method extracts the full name out of it.
+   *
+   * @param field The field from which to get the name from
```

Comment:
super nit & ditto below functions
```suggestion
   * @param field The field from which to get the name from.
```